File: 2102-find-the-middle-index-in-array/2102-find-the-middle-index-in-array.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        var totalSum = 0;
        for (var num: nums) {
            totalSum += num;
        }

        var leftSum = 0;
        for (var i = 0; i< nums.length; i++) {
            var rightSum = totalSum - leftSum - nums[i];
            if (leftSum == rightSum) {
                return i;
            }
            leftSum += nums[i];
        }
        return -1;
    }
}

File: 2102-find-the-middle-index-in-array/2102-find-the-middle-index-in-array.go
func findMiddleIndex(nums []int) int {
    totalSum := 0
    for _, num := range nums {
        totalSum += num
    }

    leftSum := 0
    for idx, num := range nums {
        rightSum := totalSum - leftSum - num
        if leftSum ==  rightSum { 
            return idx 
        }
        leftSum += num
    }
    return -1
}